CameraOverlay/data.py

```diff
@@ -0,0 +1,89 @@
+import time
+from typing import Any, Optional
+import topics as topics
+
+V2_DATA_TOPICS = [
+    str(topics.DAS.data),
+    str(topics.PowerModel.recommended_sp),
+    str(topics.PowerModel.predicted_max_speed),
+    str(topics.PowerModel.plan_name),
+]
+
+V3_MESSAGE = [
+    str(topics.DAShboard.receive_message)
+]
+
+class Data:
+
+    data_types = {
+        # DAS data
+        "power": int,
+        "cadence": int,
+        "gps": int,
+        "gps_speed": float,
+        "reed_velocity": float,
+        "reed_distance": float,
+
+        # Power model data
+        "rec_power": float,
+        "rec_speed": float,
+        "predicted_max_speed": float,
+        "zdist": float,
+        "plan_name": str,
+    }
+
+    def __init__(self):
+        self.data = {
+            # DAS data
+            "power": 0,
+            "cadence": 0,
+            "gps": 0,
+            "gps_speed": 0,
+            "reed_velocity": 0,
+            "reed_distance": 0,
+
+            # Power model data
+            "rec_power": 0,
+            "rec_speed": 0,
+            "predicted_max_speed": 0,
+            "zdist": 0,
+            "plan_name": "",
+        }
+
+        self.message_recieved_time = 0
+        self.message_duration = 5 # seconds
+        self.message = None
+
+    def load_v2_query_string(self, data: str) -> None:
```

Comment:
What happens once we have v4? v5? v6? I'm worried this class will become cluttered with functions in regards to the bike version. Already seen here by `load_v3_module_data` and `load_v3_message`.

I think it might be worth having like a base `Data` class which has a `load_data` method that needs to be implemented by other classes. Then make a `V2_Data` and a `V3_Data` class that overwrites `load_data` with their appropriate method.

Would this affect your other code?

Reply:
I'll think more about this, but the first issue that comes to mind is that right now our V2 bike uses the V3 spec for messages.

Reply:
Ahh you're right. https://github.com/monash-human-power/data-acquisition-system/blob/master/Raspi/DAS.js sends it via V3 spec. Wouldn't this mean we can get rid of the V3 spec altogether? Or should we continue using V2 Spec because it might be useful for other scripts we have?

Reply:
I actually meant V3 messages as in the messages sent from the DAShboard to the rider, I hadn't realised that `DAS.js` also uses V3 spec.

I don't think getting rid of the V2 spec entirely is worth it at this point, perhaps it can be a project for when we have less to do. I think supporting both V2 and V3 spec in the same class can be useful, especially in this intermediate stage where we're regularly using both specs simultaneously.

Having just a single class also means no configuration changes are required to switch a display from the V3 to the V2 bike, as they run the same code. I think we agreed that this would be preferable a while ago. What are your thoughts?

Reply:
I'm happy to leave it in the class now since the V3 spec is solely for camera message. However, when we start to develop an overlay that takes in V3 messages. We definitely need to create the `Data` Abstract base class. This `Data` class will simply get cluttered with different bike topics and each overlay will subscribe to both V2 and V3 topics.